UniWatch/DataAccess/ClassManager.cs

```diff
@@ -0,0 +1,206 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using UniWatch.Models;
+using System.Data.Entity;
+
+namespace UniWatch.DataAccess
+{
+    /// <summary>
+    /// Provides central access to class related information
+    /// </summary>
+    public class ClassManager : IClassManager
+    {
+        private bool disposed = false;
+        private AppDbContext _db;
+
+        /// <summary>
+        /// Creates a default ClassManager
+        /// </summary>
+        public ClassManager()
+        {
+            _db = new AppDbContext();
+        }
+
+        /// <summary>
+        /// Creates a ClassManager with the given context
+        /// </summary>
+        /// <param name="context">The context to create the manager with</param>
+        public ClassManager(AppDbContext context)
+        {
+            _db = context;
```

Comment:
Rather than having the database created here, would it be better practice to have the below code?

```
using (var context = new AppDbContext())
{
    // Your code here
}
```

[MSDN](https://msdn.microsoft.com/en-us/data/jj729737)